app/models/validations/financial_validations.rb

```diff
@@ -181,20 +181,34 @@ def validate_rent_range(record)
     return if record.startdate.blank?
 
     collection_year = record.collection_start_year
-    rent_range = LaRentRange.find_by(start_year: collection_year, la: record.la, beds: record.beds, lettype: record.lettype)
+
+    max_beds_in_rent_ranges = 4
+    validation_beds = if record.needstype == 2
+                        0
+                      else
+                        record.beds.nil? ? nil : [record.beds, max_beds_in_rent_ranges].min
+                      end
+
+    rent_range = LaRentRange.find_by(start_year: collection_year, la: record.la, beds: validation_beds, lettype: record.lettype)
```

Comment:
```suggestion

    max_beds_in_rent_ranges = 4
    validation_beds = if record.needstype == 2
                        0
                      else
                        record.beds.nil? ? nil : [record.beds, max_beds_in_rent_ranges].min
                      end

    rent_range = LaRentRange.find_by(start_year: collection_year, la: record.la, beds: record. beds_for_la_rent_range, lettype: record.lettype)
```

Do you need to repeat yourself here?

Reply:
Nope, thanks :D